bin/build.js

```diff
@@ -1,41 +1,46 @@
 /* eslint-disable no-console */
-import esbuild from 'esbuild';
+import * as esbuild from 'esbuild';
 
-const buildDirectory = 'dist';
-const production = process.env.NODE_ENV === 'production';
+// Config output
+const BUILD_DIRECTORY = 'dist';
+const PRODUCTION = process.env.NODE_ENV === 'production';
 
 // Config entrypoint files
-const entryPoints = ['src/index.ts'];
+const ENTRY_POINTS = ['src/index.ts'];
 
-/**
- * Default Settings
- * @type {esbuild.BuildOptions}
- */
-const defaultSettings = {
+// Config dev serving
+const LIVE_RELOAD = !PRODUCTION;
+const SERVE_PORT = 3000;
+
+// Create context
+const context = await esbuild.context({
   bundle: true,
-  outdir: buildDirectory,
-  minify: production,
-  sourcemap: !production,
-  target: production ? 'es2017' : 'esnext',
-  entryPoints,
-};
+  entryPoints: ENTRY_POINTS,
+  outdir: BUILD_DIRECTORY,
+  minify: PRODUCTION,
+  sourcemap: !PRODUCTION,
+  target: PRODUCTION ? 'es2019' : 'esnext',
+  inject: LIVE_RELOAD ? ['./bin/live-reload.js'] : undefined,
+  define: {
+    SERVE_PORT: `${SERVE_PORT}`,
+  },
+});
 
-// Files building
-if (production) {
-  esbuild.build(defaultSettings);
+// Build files in prod
+if (PRODUCTION) {
+  await context.rebuild();
+  context.dispose();
 }
 
-// Files serving
+// Watch and serve files in dev
 else {
-  esbuild
-    .serve(
-      {
-        servedir: buildDirectory,
-        port: 3000,
-      },
-      defaultSettings
-    )
-    .then((server) => {
-      console.log(`Serving at http://localhost:${server.port}`);
+  await context.watch();
+  await context
+    .serve({
+      servedir: BUILD_DIRECTORY,
+      port: SERVE_PORT,
+    })
+    .then(({ port }) => {
+      console.log(`Serving at http://localhost:${port}`);
```

Comment:
@alexiglesias93 @aexshafii @robertkibet I've been using this for easy reference, what do you think?

``// Log all served files for easy reference.
      console.log(
        'Serving built files at:',
        entryPoints.map((path) => ({
          url: `http://localhost:${server.port}/${path.replace('src/', '')}`,
        }))
      );``
      
Which should print something like this:
      
![image](https://user-images.githubusercontent.com/1149952/213453813-ef6313f0-d131-4339-b2af-a043279b2fb8.png)

Reply:
@erikmejias I love it, I'll add this

Reply:
[52b4a83](https://github.com/finsweet/developer-starter/pull/25/commits/52b4a838910e9aca6913c72549e735e3b04ddf30)